File: Java/DesignPatterns/src/test/java/designPatterns/structural/decorator/MainDecoratorTest.java
package designPatterns.structural.decorator;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MainDecoratorTest {
  private Component concreteCom;
  private Component decorateCom;

  @BeforeEach
  void setUp() {
    concreteCom = new ConcreteComponent();
    decorateCom = new ConcreteDecorator(new ConcreteComponent());
  }

  @Test
  void equalsTest() {
    assertThat(concreteCom.assemble()).isEqualTo("Concrete Component");
    assertThat(decorateCom.assemble()).isEqualTo("Concrete Component\nAdding Decorator");
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/structural/decorator/MainDecoratorTest.kt
package structural.decorator

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MainDecoratorTest {
  private lateinit var concreteCom: Component
  private lateinit var decorateCom: Component

  @BeforeEach
  internal fun setUp() {
    concreteCom = ConcreteComponent()
    decorateCom = ConcreteDecorator(ConcreteComponent())
  }

  @Test
  internal fun equalsTest() {
    assertThat(concreteCom.assemble()).isEqualTo("Concrete Component")
    assertThat(decorateCom.assemble()).isEqualTo("Concrete Component\nAdding Decorator")
  }
}